### PR TITLE
refactor(controls): use Class for GlobeControls.

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
Use Class for `GlobeControls`.

Waiting #1266

### CODE REVIEW

File: src/Controls/GlobeControls.js [hunk 1]
```diff
@@ -131,8 +130,28 @@ export const CONTROL_EVENTS = {
...
+function getRangeFromScale(scale, pitch, fov, height) {
+    // Screen pitch, in millimeters
+    pitch = (pitch || 0.28) / 1000;
+    fov = THREE.Math.degToRad(fov);
+    // Invert one unit projection (see getDollyScale)
+    const range = pitch * height / (scale * 2 * Math.tan(fov * 0.5));
+
+    return range;
+}
+
```
Comment (zarov) [inline src/Controls/GlobeControls.js:154]: Maybe you could replace this by `View#getScaleFromDistance`, or at least use it.

File: src/Controls/GlobeControls.js [hunk 2]
```diff
@@ -156,104 +175,172 @@ let previous;
...
+        this.player = new AnimationPlayer();
+        this._view = view;
+        this.camera = view.camera.camera3D;
+        this.domElement = view.mainLoop.gfxEngine.renderer.domElement;
+
+        // State control
+        this.states = new StateControl();
+        this.state = this.states.NONE;
+
+        this.getStates = () => this.states;
```
Comment (zarov) [inline src/Controls/GlobeControls.js:190]: Is it necessary ?

File: src/Controls/GlobeControls.js (hunk 1 above)
Comment (zarov) [reply src/Controls/GlobeControls.js:154]: It seems you can also replace things like `getRange` by `View#getDistanceFromCamera`. There seems to be quite a lot of method that are giving the same result while doing different things, so maybe take a look into that if you have time (between `View`, `CameraUtils` and `GlobeControls` for instance).

**Review by zarov — COMMENTED** [review]
Nothing much to say, thanks for continuing this cleaning ! Waiting for the test to approve.

File: src/Controls/GlobeControls.js (hunk 2 above)
Comment (gchoqueux) [reply src/Controls/GlobeControls.js:190]: no, I will remove in next clean PR.

File: src/Controls/GlobeControls.js (hunk 1 above)
Comment (gchoqueux) [reply src/Controls/GlobeControls.js:154]: Yes, I'll create a new PR to re-factorize and I only use the class in this PR.

**Review by zarov — APPROVED** [review]

**Pull request merged**